numpy/core/fromnumeric.py

```diff
@@ -1938,12 +1938,12 @@ def compress(condition, a, axis=None, out=None):
     return _wrapfunc(a, 'compress', condition, axis=axis, out=out)
 
 
-def _clip_dispatcher(a, a_min, a_max, out=None):
+def _clip_dispatcher(a, a_min, a_max, out=None, **kwargs):
```

Comment:
I agree - what's left of clip is just a ufunc selector based on argument presence, there's no reason to allow customization of that.

Reply:
Well, the current promise is that *everything* directly exposed by NumPy [gets its own dispatcher](http://www.numpy.org/neps/nep-0018-array-function-protocol.html#implementations-in-terms-of-a-limited-core-api), either via `__array_function__` or `__array_ufunc__`. This PR means that `np.clip` now uses one of three ufuncs, one of which (`um.clip`) isn't even publicly exposed. So I don't think we can get rid of this dispatcher yet.

Reply:
Should we consider `np.core.umath.clip` sufficient exposure? It can never replace `np.clip` in the top namespace and IMO there's no clearer name available

Reply:
I'd think that `np.core.umath` is sufficient exposure (indeed, in astropy we check our ufunc coverage by enumerating what is in there). The larger question might be whether this multi-dispatch would still need an override. To me, that seems a bit odd, and counter to the idea that `__array_function__` is only meant to cover what isn't covered by better matched overrides.

p.s.  An analogy here may be `ndarra.__pow__`, which gives different `ufuncs` depending on the argument (though of course subclasses can override that...).

Reply:
The rest of `np.core` is not considered public API so I don't think that's enough exposure. Maybe we could put it as an attribute on the function, e.g., `np.clip.ufunc`?

I'm also reluctant to do this on an ad-hoc basis. I don't think it's worth the complexity of having exactly *one* special function that is neither a ufunc nor with a direct override via `__array_function__`. We don't need to change this all at once but at least we should have a rule for going forward that clearly applies to more than only a few functions.

Reply:
OK, perhaps not that simple. Maybe for now best to finish this PR without removing the `__array_function__` override?